src/Type/ObjectType.php

```diff
@@ -706,6 +706,10 @@ public function toArrayKey(): Type
 
 	public function toCoercedArgumentType(bool $strictTypes): Type
 	{
+		if (!$strictTypes) {
+			return TypeCombinator::union($this, $this->toString());
```

Comment:
If the object isn't a Stringable, this should just return `$this`. But instead it will return ErrorType.

Also I'm not sure what happens with `BcMath\Number` in strict_types=1 or 0.

Reply:
added tests with links to 3v4l.org